src/SharedUtilities/Domains.jl

```diff
@@ -408,14 +408,96 @@ are expected to be sorted in ascending order.
 struct RootDomain{FT} <: AbstractVegetationDomain{FT}
     "The depth of the root tips, in meters"
     root_depths::Vector{FT}
+    "The number of stem compartments for the plant"
+    n_stem::Int64
+    "The number of leaf compartments for the plant"
+    n_leaf::Int64
     "The height of the stem, leaf compartments, in meters"
     compartment_heights::Vector{FT}
+    "The type (stem or leaf) each compartment is"
+    compartment_labels::Vector{Symbol}
+
+    function RootDomain(
+        root_depths::Vector{FT},
+        n_stem::Int64,
+        n_leaf::Int64,
+        delta_z::FT,
```

Comment:
try "\Delta" followed by a tab. should give you the greek symbol Delta. 
then we can use \Delta z as the variable name.